### ISSUE TITLE
Broken when using base_path

### ISSUE THREAD

**Issue by laurensvane** [netbox-community/netbox-docker#938]
### Current Behavior

I tried to setup base_path in the netbox.env because the netbox is served from behind a proxy under a subpath on the server.
When i setup base_path to be 'netbox/'
I can reach the web interface, but the netbox-housekeeping and netbox-worker container won't start.
In the netbox container logs i get the following api errors:
127.0.0.1 - - [31/Jan/2023:11:23:06 +0000] "GET /api/ HTTP/1.1" 404 11189 "-" "curl/7.81.0"
Not Found: /api/

When i remove the base_path variable, all containers start without errors, but i can't reacht the web interface of netbox through the proxy.

### Expected Behavior

I expected the internal communication of the containers to get the base_path and adjust their calls to the netbox container as it should.

### Docker Compose Version

Docker Compose Version 1.29.2

### Docker Version

```text
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

### The git Revision

39c7de4af41703108f8b3af7274bc65813c41799

### The git Status

```text
On branch release
Your branch is up to date with 'origin/release'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        netbox.env.override

nothing added to commit but untracked files present (use "git add" to track)
```

### Startup Command

docker-compose up

### NetBox Logs

```text
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
💡 Superuser Username: admin, E-Mail: admin@example.com
✅ Initialisation is done.
⏳ Waiting for control socket to be created... (1/10)
2023/01/31 14:06:10 [warn] 7#7 Unit is running unprivileged, then it cannot use arbitrary user and group.
2023/01/31 14:06:10 [info] 7#7 unit 1.27.0 started
2023/01/31 14:06:10 [info] 21#21 discovery started
2023/01/31 14:06:10 [notice] 21#21 module: python 3.10.4 "/usr/lib/unit/modules/python3.10.unit.so"
2023/01/31 14:06:10 [info] 7#7 controller started
2023/01/31 14:06:10 [notice] 7#7 process 21 exited with code 0
2023/01/31 14:06:10 [info] 23#23 router started
2023/01/31 14:06:10 [info] 23#23 OpenSSL 3.0.2 15 Mar 2022, 30000020
⚙️ Applying configuration from /etc/unit/nginx-unit.json
2023/01/31 14:06:11 [info] 27#27 "netbox" prototype started
2023/01/31 14:06:11 [info] 28#28 "netbox" application started
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
✅ Unit configuration loaded successfully
2023/01/31 14:06:13 [notice] 7#7 process 19 exited with code 0
2023/01/31 14:06:23 [info] 52#52 "netbox" application started
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
Not Found: /api/
127.0.0.1 - - [31/Jan/2023:14:06:27 +0000] "GET /api/ HTTP/1.1" 404 11188 "-" "curl/7.81.0"
```

### Content of docker-compose.override.yml

```yaml
version: '3.4'
services:
  netbox:
    ports:
      - 9001:8080
    build:
      context: .
      dockerfile: Dockerfile-Plugins
    restart: unless-stopped
  netbox-worker:
    restart: unless-stopped
  netbox-housekeeping:
    restart: unless-stopped
  postgres:
    image: postgres:14-alpine
    restart: unless-stopped
  redis:
    restart: unless-stopped
  redis-cache:
    restart: unless-stopped

    #image: netbox:latest-plugins
#  netbox-worker:
    #image: netbox:latest-plugins
#  postgres:
#    image: postgres:13-alpine

networks:
  default:
    ipam:
      driver: default
      config:
        - subnet: 10.10.230.0/24
```

**Comment by jnovak-netsystemcz** [comment]
Hi, I can confirm your findings and I'm touching your issue too.

I found that the issue is that content is provided by NGINX unit. Its config file is /etc/unit/nginx-unit.json and this config file is fixed. It do not take into account BASE_PATH variable. Therefore if I modify the file (e.g. volume mapping of the file) from:

```
...
  "routes": [
    {
      "match": {
        "uri": "/static/*"
      },
      "action": {
        "share": "/opt/netbox/netbox${uri}"
      }
    },
...

```

to (my prefix is netbox/):

```
...
  "routes": [
    {
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox${uri}"
      }
    },
...
```

It allows me to login to netbox. The issue is that once I login, it starts to redirect over and over to /netbox URI therefore it do not work again :-(

I found no solution to redirect loop yet...
